File: app/src/main/java/com/blueskybone/arkscreen/base/preference/Preference.java
package com.blueskybone.arkscreen.base.preference;

/*
 * The code for data persistence refers to the following project
 * https://github.com/easybangumiorg/EasyBangumi
 * */

public interface Preference <T> {
    String key();
    T get();
    void set(T value);

    T defaultValue();
    Boolean isSet();

    void delete();
}

File: app/src/main/java/com/blueskybone/arkscreen/preference/preference/Preference.kt
package com.blueskybone.arkscreen.preference.preference

/**
 *   Created by blueskybone
 *   Date: 2025/1/7
 */
interface Preference<T> {
    fun key(): String
    fun get(): T
    fun set(value: T)

    fun defaultValue(): T
    fun isSet(): Boolean

    fun delete()
}